diff --git a/CMakeLists.txt b/CMakeLists.txt
index f0e33cd..c91238a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,8 +2,6 @@ cmake_minimum_required(VERSION 2.8)
 
 project(litehtml CXX)
 
-add_subdirectory(src/gumbo)
-
 set(SOURCE_LITEHTML
     src/background.cpp
     src/box.cpp
@@ -43,16 +41,79 @@ set(SOURCE_LITEHTML
     src/web_color.cpp
 )
 
-add_library(${PROJECT_NAME} ${SOURCE_LITEHTML})
+set(HEADER_LITEHTML
+    include/litehtml.h
+    src/attributes.h
+    src/background.h
+    src/borders.h
+    src/box.h
+    src/context.h
+    src/css_length.h
+    src/css_margins.h
+    src/css_offsets.h
+    src/css_position.h
+    src/css_selector.h
+    src/document.h
+    src/element.h
+    src/el_anchor.h
+    src/el_base.h
+    src/el_before_after.h
+    src/el_body.h
+    src/el_break.h
+    src/el_cdata.h
+    src/el_comment.h
+    src/el_div.h
+    src/el_font.h
+    src/el_image.h
+    src/el_link.h
+    src/el_para.h
+    src/el_script.h
+    src/el_space.h
+    src/el_style.h
+    src/el_table.h
+    src/el_td.h
+    src/el_text.h
+    src/el_title.h
+    src/el_tr.h
+    src/html.h
+    src/html_tag.h
+    src/iterators.h
+    src/media_query.h
+    src/os_types.h
+    src/style.h
+    src/stylesheet.h
+    src/table.h
+    src/types.h
+    src/utf8_strings.h
+    src/web_color.h
+)
+
+find_path(gumbo_INCLUDE gumbo.h REQUIRED)
+find_library(gumbo_LIB gumbo REQUIRED)
+
+include_directories(${gumbo_INCLUDE})
+
+add_library(${PROJECT_NAME} SHARED ${SOURCE_LITEHTML})
+target_link_libraries(${PROJECT_NAME} ${gumbo_LIB})
 
 set_target_properties(${PROJECT_NAME} PROPERTIES
     CXX_STANDARD 11
     C_STANDARD 99
+    PUBLIC_HEADER "${HEADER_LITEHTML}"
 )
 
 # Export litehtml includes.
 target_include_directories(${PROJECT_NAME} PUBLIC src)
 target_include_directories(${PROJECT_NAME} PUBLIC include)
 
-# Gumbo
-target_link_libraries(${PROJECT_NAME} PUBLIC gumbo)
+# install and export
+install(TARGETS ${PROJECT_NAME}
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+    PUBLIC_HEADER DESTINATION include/litehtml
+)
+#install(FILES ${PUBLIC_HEADERS_LITEHTML} DESTINATION include/litehtml)
+
+
+
